columnflow/config_util.py

```diff
@@ -83,6 +84,36 @@ def get_shifts_from_sources(config: od.Config, *shift_sources: str) -> list[od.S
     )
 
 
+def create_category_id(
+    config: od.Config,
+    category_name: str,
+    hash_len: int = 7,
+) -> int:
+    """
+    Creates a unique id for a :py:class:`order.Category` named *category_name* in a
+    :py:class:`order.Config` object *config* and returns it. Internally,
+    :py:func:`law.util.create_hash` is used which receives *hash_len*.
+    """
+    h = law.util.create_hash((config.name, config.id, category_name), l=hash_len)
+    return int(h, base=16)
```

Comment:
would be nice if we could add an offset here (of 10^8?) to avoid overlap with category IDs manually set by the user

Reply:
Yup, added that, plus the option to add a custom *salt* value.